iac-extensions/docker/src/test/java/org/sonar/iac/docker/symbols/ArgumentResolutionTest.java

```diff
@@ -166,6 +172,65 @@ void shouldNotDeadLoopWhenResolvingSelfAssignedVariable() {
     assertThat(resolution.status()).isEqualTo(UNRESOLVED);
   }
 
+  @ParameterizedTest
+  @MethodSource
+  void shouldCorrectlyHandleQuotes(String input, String expectedSecondArgument) {
+    File file = parseFileAndAnalyzeSymbols(input);
+
+    List<ArgumentResolution> argumentResolutions = CheckUtils.resolveInstructionArguments(TreeUtils.firstDescendant(file, RunInstruction.class).get());
+    ArgumentResolution stringArgument = argumentResolutions.get(1);
+
+    assertThat(stringArgument.value()).isEqualTo(expectedSecondArgument);
+  }
+
+  static Stream<Arguments> shouldCorrectlyHandleQuotes() {
+    return Stream.of(
+      // argument of `echo` is StringLiteral
+      "RUN echo foo#foo",
+      "RUN echo \"foo\"#\"foo\"",
+      "RUN echo 'foo'#'foo'",
+      "RUN echo '\"foo\"'#'\"foo\"'",
+      "RUN echo \"\"foo\"\"#\"\"foo\"\"",
+      "RUN echo \"'foo'\"#\"'foo'\"",
+      // argument of `echo` is ExpandableStringLiteral
+      "RUN echo $FOO#foo",
+      "RUN echo \"$FOO\"#\"foo\"",
+      // single-quoted variables are not substituted by Docker, so we shouldn't resolve them
+      "RUN echo '$FOO'#'$FOO'",
+      "RUN echo \"\"$FOO\"\"#\"\"foo\"\"",
+      "RUN echo ${FOO}#foo",
+      "RUN echo \"${FOO}\"#\"foo\"",
+      "RUN echo '${FOO}'#'${FOO}'",
+      "RUN echo \"\"${FOO}\"\"#\"\"foo\"\"",
+      // in ExecForm all arguments in an array are ExpandableStringLiterals
+      "RUN [\"echo\", \"$FOO\"]#foo",
+      "RUN [\"echo\", \"'$FOO'\"]#'foo'",
+      "RUN [\"echo\", \"${FOO}\"]#foo",
+      "RUN [\"echo\", \"'${FOO}'\"]#'foo'",
+      // in ExecForm quotes have to be escaped and we are not un-escaping them
+      "RUN [\"echo\", \"\\\"$FOO\\\"\"]#\\\"foo\\\"",
+      "RUN [\"echo\", \"\\\"${FOO}\\\"\"]#\\\"foo\\\"")
+      .map(s -> "FROM scratch\nARG FOO=\"foo\"\n" + s)
+      .map(s -> s.split("#"))
+      .map(Arguments::of);
```

Comment:
Interesting idea how to handle it, but maybe try this:
```
return Stream.of(
      // argument of `echo` is StringLiteral
      Arguments.of("RUN echo foo", "foo"), ....
```
And the logic from line 213 can be moved to the test method. Then it is much clearer what is input and expected result.

Reply:
I agree that it can be simplified, but I didn't want to repeat `Arguments.of` on every line. And with static import simple `of()` won't be very clear IMO